---
title: "Quantitative analysis ICB"
author: "Franziska Lang"
date: "`r Sys.Date()`"
output: 
  html_document:
    code_folding: hide
    toc: true
    toc_depth: 3    
    toc_float: true
    highlight: pygments
editor_options: 
  chunk_output_type: console
knit: (
  function(inputFile, encoding) { 
    rmarkdown::render(
      inputFile, 
      encoding = encoding, 
      output_file = file.path(dirname(inputFile), "icb_data_overview_expression_v2.html")) }
      )
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(warning = FALSE, message = FALSE, echo = F)

# ADJUST PATHS HERE 
setwd("/projects/SUMMIT/WP1.2/Literature_Cohorts/data_analysis/code/miles_publication/")
path_to_plots <- "plots/"

date_today <- gsub("-", "", Sys.Date())
library(tidyverse)
library(ggpubr)
library(gridExtra)
library("wesanderson")
source("data_analysis/functions_icb2neo.R")
source( "import_data.R")
library(ROCR)
theme_set(theme_bw())


col_mutation <- wes_palette("FantasticFox1",3)
col_expressed <- wes_palette("Chevalier1",2)
col_entity <- c("#E7CDC2", "#80A0C7", "#394165")


export_plots <- FALSE
export_table <- FALSE

```

```{setup, include=FALSE}


# technical issues with iCaM --> SNVs + INDELs will be counted as NA 
# Hugo Pt 8 
# McDermott Pt164370145747331
# 


# no SNVs/INDELS were detected for the following patients why here we can count 0
# Riaz pre Pt24
# McDermott Pt204886393976620

```

```{r setup, include=FALSE}
#neoantigen features
features_full <- c("Best_rank_MHCI_score", "Best_rank_MHCII_score", "rnaExpression", "Pathogensimiliarity_MHCI_9mer", "Pathogensimiliarity_MHCII",
                     "Selfsimilarity_MHCI", "Selfsimilarity_MHCII", "rnaVariantAlleleFrequency", "Amplitude_MHCII_rank", "Amplitude_MHCI_affinity",
                     "DAI_MHCI_affinity", "Dissimilarity_MHCI", "Generator_rate_MHCI", "Generator_rate_MHCII", "IEDB_Immunogenicity_MHCI", 
                     "IEDB_Immunogenicity_MHCII","MixMHC2pred_best_rank", "MixMHCpred_best_rank", "PHBR_I", "PHBR_II", "vaxrank_binding_score",
                     "Hex_alignment_score_MHCI", "Hex_alignment_score_MHCII", "Neoag_immunogenicity" , "Priority_score", "Recognition_Potential_MHCI_9mer",
                   "Tcell_predictor_score", "vaxrank_total_score", "PRIME_best_rank")

```



## AIM

The aim is to provide a quantitative overview over the ICB data. Here, only expressed neoantigen candidates are considered  
The following cohorts were used:  

```{r ,  echo=FALSE, message=FALSE, warning = FALSE}

df_cohort <- tibble(
  cohort = names(dat_pat),
  "# patients" = sapply(dat_pat, length)
)

knitr::kable(df_cohort)
```


```{r ,  echo=FALSE, message=FALSE, warning = FALSE}

# remove SD
dat_fusions_transformed$riaz_pre$response <- dat_fusions_transformed$riaz_pre$Response
dat_fusions_transformed$miao$response <- dat_fusions_transformed$miao$best_RECIST
dat_indel_transformed$miao$response <- dat_indel_transformed$miao$best_RECIST
dat_transformed$miao$response <- dat_transformed$miao$best_RECIST

dat_fusions_transformed <- lapply(dat_fusions_transformed, function(x){
  x%>% filter(response != "SD")
})

dat_indel_transformed <- lapply(dat_indel_transformed, function(x){
  x%>% filter(response != "SD")
})

dat_transformed <- lapply(dat_transformed, function(x){
  x%>% filter(response != "SD")
})


```


## 1 Quantitative overview

### 1.1 Number of neoantigen candidates per patient

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# number of all possible neoantigen candidates from different sources per patient
number_snvs <- get_numbers(dat_transformed, nam = "snv")
number_indels <- get_numbers(dat_indel_transformed, nam = "INDEL")
number_fusion_genes <- get_numbers(dat_fusions_transformed, nam = "fusion gene")

numbers_all <- merge_counts(number_snvs, number_indels, number_fusion_genes)

```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# get numbers of expressed neoantigen candidates
# variant allele expression > 0
dat_pat_expressed <- lapply(dat_transformed, filter_expressed)
dat_indel_pat_expressed <- lapply(dat_indel_transformed, filter_expressed)
dat_fusions_pat_expressed <- lapply(dat_fusions_transformed, function(y) filter_expressed(y,"fusion gene"))

# number of neoantigen candidates from different sources per patient
number_snvs_expressed <- get_numbers(dat_pat_expressed, nam = "snv")
number_indels_expressed <- get_numbers(dat_indel_pat_expressed, nam = "INDEL")
number_fusion_genes_expressed <- get_numbers(dat_fusions_pat_expressed, nam = "fusion gene")

# combined number expressed candidates
numbers_expressed <- merge_counts(number_snvs_expressed, number_indels_expressed, number_fusion_genes_expressed)

```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# get numbers of neoantigen candidates that are predicted to bind to MHC 
dat_pat_binding <- lapply(dat_transformed, filter_mhc_binding)
dat_indel_pat_binding <- lapply(dat_indel_transformed, filter_mhc_binding)
dat_fusions_pat_binding <- lapply(dat_fusions_transformed, function(y) filter_mhc_binding(y,"fusion gene"))

# number of neoantigen candidates from different sources per patient
number_snvs_binding <- get_numbers(dat_pat_binding, nam = "snv")
number_indels_binding <- get_numbers(dat_indel_pat_binding, nam = "INDEL")
number_fusion_genes_binding <- get_numbers(dat_fusions_pat_binding, nam = "fusion gene")

# combined number expressed candidates
numbers_binding <- merge_counts(number_snvs_binding, number_indels_binding, number_fusion_genes_binding)

```


The boxplot shows for each cohort the predicted candidate count per patient in respect to the mutation class (expressed and non-expressed are considered).  

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# tall-skinny data frame of all cohorts
# NOTE: NA is counted as 0 --> while this holds true for most of the patients-mut class combination there were a very few patients for which NA should derive from a MyMut error that could not be solved
DF2 <- transfrom_count2tall(numbers_all)
DF2_plot <- DF2 %>%
  filter(!class == "all")


## FIGURE 2A
if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_boxplot_number_all_candidates_class.pdf")
  pdf(nam, width = 10, height = 6)
}

xlabs <- paste(levels(DF2_plot$cohort),"\n(n=",table(DF2_plot$cohort[which(DF2_plot$class == "SNV")]),")",sep="")

DF2_plot %>%
  ggplot(aes(x = cohort, y = count + 1, fill = class)) +
  geom_boxplot(outlier.fill = NULL, outlier.shape = 21) +
  scale_y_continuous(trans ='log10') +
  scale_fill_manual(values = col_mutation) + 
  scale_color_manual(values =col_mutation) +
  theme(panel.grid.major = element_blank())+
  scale_x_discrete(labels=xlabs)


if(export_plots){
  dev.off()
  #ggsave(nam, p,  width=6, height=10)
}  


```

Proportions of mutation source for neoantigen candidates per patient:  

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}
 
# substitute NAs with 0 in the main data frame and calculate percentage 
number_summary <- lapply(names(numbers_all), function(x){
  technical_problems <- c("hugo" = "Pt8", "mcdermott_Atezo" =  "Pt164370145747331", "riaz_pre" = "no", "vanallen" = "no")
  numbers_all[[x]] %>%
    mutate(snv = ifelse(is.na(snv) & patientIdentifier != technical_problems[x], 0, snv )) %>%
    mutate(INDEL = ifelse(is.na(INDEL) & patientIdentifier != technical_problems[x], 0, INDEL )) %>%
    mutate(all = rowSums(.[2:4], na.rm = FALSE))%>%
    mutate(percentage_snv = snv/all, 
           percentage_indel = INDEL/all,
           percentage_fusion = `fusion gene`/all) %>%
    filter(!is.na(percentage_snv))
  
})
names(number_summary) <- names(number_snvs)

## FIGURE 2B
if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_proportion_neoantigen_class.pdf")
  pdf(nam)
}
par(mar=c(10,4,4,4))
lapply(names(number_summary), function(x){

  make_proportion_barplot(number_summary[[x]], main = x)
})
if(export_plots){
  dev.off()
}    
```


The following boxplot show the number all predicted neoantigen candidates per patient and the number of expressed neoantigen candidates for SNVs, INDELs and fusion genes.  


```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# USE THIS AS SUPPLEMENTAL FIGURE
DF2_expressed <- transfrom_count2tall(numbers_expressed)
DF2_expressed_plot <- DF2_expressed %>%
  filter(class != "all")
DF_all_expressed <- bind_rows(list(all = DF2_plot, expressed = DF2_expressed_plot), .id = "category")

# compare numbers of all predicted and expressed neoantigen candidates
if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_comparison_all_expressed.pdf")
  pdf(nam)
}

boxplot_all_selected(DF_all_expressed, main = "all")
boxplot_all_selected_facet(DF_all_expressed)

if(export_plots){
  dev.off()
}  

```


The following boxplot show the number all predicted neoantigen candidates per patient and the neoantigen candidates that generate at least one MHC binding epitope for SNVs, INDELs and fusion genes.  

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# USE THIS AS SUPPLEMENTAL FIGURE
DF2_binding <- transfrom_count2tall(numbers_binding)
DF2_binding_plot <- DF2_binding %>%
  filter(class != "all")
DF_all_binding <- bind_rows(list(all = DF2_plot, expressed = DF2_binding_plot), .id = "category")

# compare numbers of all predicted and binding neoantigen candidates

if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_comparison_all_binding.pdf")
  pdf(nam)
}

boxplot_all_selected(DF_all_binding, main = "all")
boxplot_all_selected_facet(DF_all_binding)
if(export_plots){
  dev.off()
}  

```


### 1.2 Neoantigen candidate quantity and response to ICB

The following boxplots compare the neoantigen candidate load (without any filter) between responding and non-responding patients. 

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

#cohorts_with_rna <- c("hugo", "vanallen", "riaz_pre", "riaz_on", "mcdermott_Atezo", "mcdermott_Atezo+Bev")
cohorts_with_rna <- c("hugo", "vanallen", "riaz_pre",  "mcdermott_Atezo")

# prepare a response data-frame
response_snv <- lapply(dat_transformed,function(x){
  x%>%
    select(patientIdentifier, binary_response) %>%
    distinct
})
response_fusions <- lapply(dat_fusions_transformed,function(x){
  x%>%
    select(patientIdentifier, binary_response) %>%
    distinct
})

response_df <- response_snv
df_resp <- lapply(names(response_snv), function(cohort){
  if(cohort %in% cohorts_with_rna){
    res <- response_snv[[cohort]] %>%
    full_join(response_fusions[[cohort]], by = "patientIdentifier")%>%
    mutate(binary_response = ifelse(is.na(binary_response.x), binary_response.y , ifelse(is.na(binary_response.y), binary_response.x, binary_response.x) )) %>%
      select(patientIdentifier, binary_response)
  }else{
    response_snv[[cohort]]%>%
      select(patientIdentifier, binary_response)
  }
})

names(df_resp) <- names(response_snv)

df_resp_df <- bind_rows(df_resp, .id = "cohort")

nice_names <- c("Van Allen","Hugo", "Riaz", "Miao", "McDermott")
names(nice_names) <- c( "vanallen" ,"hugo"  ,"riaz_pre" ,"miao" ,"mcdermott_Atezo")
df_resp_df <- df_resp_df %>%
   mutate(cohort = nice_names[cohort]) %>%
  mutate(cohort_pat = paste0(cohort,"_" ,patientIdentifier)) 


```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp <- DF2 %>%
  mutate(cohort_pat = paste0(cohort,"_" ,patientIdentifier)) %>%
  left_join(df_resp_df, by = "cohort_pat")

```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# boxplot showing number of neoantigen candidates load between responder and non-responder
col_response <- c(no=rgb(12,44,132, 255, maxColorValue = 255),
                  yes = rgb(178,34,34, 250 , maxColorValue = 255))

if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_boxplot_response_number_all_candidates.pdf")
  pdf(nam)
}    

DF2_resp %>%
  make_boxplot_response_facet()

if(export_plots){
  dev.off()
}    

pval_all <- compare_means(count ~ binary_response, data = DF2_resp, group.by = c("class", "cohort.x"))
pval_all_all_cohorts <- compare_means(count ~ binary_response, data = DF2_resp, group.by = c("class"))
pval_all_all_cohorts$cohort.x <- "all"

pval_all <- bind_rows(pval_all, pval_all_all_cohorts)

```



### 1.3 Expressed neoantigen candidate quantity and response to ICB

The following boxplots compare the load with expressed number of neoantigen candidates between responding and non-responding patients. 

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# merge response and number of neoantigen candidates
DF2_expresed_resp <- DF2_expressed %>%
  mutate(cohort_pat = paste0(cohort,"_" ,patientIdentifier)) %>%
  left_join(df_resp_df, by = "cohort_pat")

pval_expressed <- compare_means(count ~ binary_response, data = DF2_expresed_resp, group.by = c("class", "cohort.x"))
pval_expressed_all_cohorts <- compare_means(count ~ binary_response, data = DF2_expresed_resp, group.by = c("class"))
pval_expressed_all_cohorts$cohort.x <- "all"

pval_expressed <- bind_rows(pval_expressed, pval_expressed_all_cohorts)


```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}


if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_boxplot_response_number_expressed_candidates.pdf")
  pdf(nam)
}

DF2_expresed_resp %>%
  make_boxplot_response_facet()

if(export_plots){
  dev.off()
}    


```


### 1.3 Number of neoantigen candidatese meeting defined thresholds and the response to ICB

Here, the neoantigen candidate load is estimated by considering only candidates that harbour MHC I or MHC II binding capabilities meeting a given threshold. 

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# neoantigen candidate load at different MHC biniding thresholds 

feature1 = "Best_affinity_MHCI_score"
feature2 = "Best_affinity_MHCII_score"

count1000 <- neoag_qual_class_count(dat_transformed, dat_indel_transformed, 
                            dat_fusions_transformed, feature1 = feature1, feature2 = feature2, thres = 1000)
pval1000 <- neoag_qual_class_pval(count1000)
count500 <- neoag_qual_class_count(dat_transformed, dat_indel_transformed, 
                            dat_fusions_transformed, feature1 = feature1, feature2 = feature2, thres = 500)
pval500 <- neoag_qual_class_pval(count500)
count50 <- neoag_qual_class_count(dat_transformed, dat_indel_transformed, 
                            dat_fusions_transformed, feature1 = feature1, feature2 = feature2, thres = 50)
pval50 <- neoag_qual_class_pval(count50)

```




```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# count expressed neoantigen candidates with best MHC I/II affinity < 50, 500, 1000 nM

dat_transformed_expressed <- lapply(dat_pat_expressed, bind_rows, .id = "cohort")
dat_indel_transformed_expressed <- lapply(dat_indel_pat_expressed, bind_rows, .id = "cohort")
dat_fusions_transformed_expressed <- lapply(dat_fusions_pat_expressed, bind_rows, .id = "cohort")

# MHC < 1000
count1000_expressed <- neoag_qual_class_count(dat_transformed_expressed, dat_indel_transformed_expressed, 
                            dat_fusions_transformed_expressed, feature1 = feature1, feature2 = feature2, thres = 1000)
pval1000_expressed <- neoag_qual_class_pval(count1000_expressed)
# MHC < 500
count500_expressed <- neoag_qual_class_count(dat_transformed_expressed, dat_indel_transformed_expressed, 
                            dat_fusions_transformed_expressed, feature1 = feature1, feature2 = feature2, thres = 500)
pval500_expressed <- neoag_qual_class_pval(count500_expressed)
# MHC < 50
count50_expressed <- neoag_qual_class_count(dat_transformed_expressed, dat_indel_transformed_expressed, 
                            dat_fusions_transformed_expressed, feature1 = feature1, feature2 = feature2, thres = 50)
pval50_expressed <- neoag_qual_class_pval(count50_expressed)



```

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}
pval_all <- pval_all %>%
  dplyr::rename(cohort = cohort.x)
pval_expressed <- pval_expressed %>%
  dplyr::rename(cohort = cohort.x)


pvals_list <- bind_rows(list(all = pval_all, all_expressed = pval_expressed, 
                             MHC1000 = pval1000, MHC500 = pval500, MHC50 =pval50,
                             MHC1000_expressed = pval1000_expressed, MHC500_expressed = pval500_expressed, MHC50_expressed =pval50_expressed ), .id = "method")

col_pal <- c( RColorBrewer::brewer.pal(5, "GnBu")[3:5], "grey90") 

  

## Supplement Figure 1


S1<- pvals_list %>%
  mutate(p.signif = factor(p.signif)) %>%
  mutate(method = factor(method, levels = rev(c("all", "all_expressed", "MHC1000",  "MHC1000_expressed", "MHC500","MHC500_expressed", "MHC50", "MHC50_expressed")))) %>%
  ggplot(aes(cohort, method, color = p.signif)) +
  geom_point(alpha = 1 , size = 8) + 
  facet_wrap(vars(class)) + 
  scale_color_manual(values =col_pal)+
  theme(axis.text.x = element_text(angle = 90, vjust = 0.5, hjust=1)) 

if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_ICB_dot_plot_pvals_count_methods.pdf")
  pdf(nam, height = 6, width = 10)
}
S1 
if(export_plots){
  dev.off()
}


```


```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# some boxplot to validate finding from dot plot 
if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_ICB_selected_count_response_correl.pdf")
  pdf(nam)
}

# ALL --> SNV 
p1 <- DF2_resp %>% 
  filter(class == "SNV") %>%
  make_boxplot_response(main = "all cohorts : SNV")

p2 <- count50 %>%
  filter(class == "SNV") %>%
  make_boxplot_response(main = "all cohorts : SNV : MHC < 50 ")
  
p3<- count50_expressed %>%
  filter(class == "SNV") %>%
  make_boxplot_response(main = "all cohorts : SNV : expressed + MHC < 50")



# ALL --> INDEL 
p4 <- count50 %>%
  filter(class == "INDEL") %>%
  make_boxplot_response(main = "all : INDEL : MHC < 50 ")
  
p5 <- count50_expressed %>%
  filter(class == "INDEL") %>%
  make_boxplot_response(main = "all : INDEL :  expressed + MHC < 50")

# RIAZ INDEL 
p6 <- count50 %>%
  filter(cohort == "riaz_pre" & class == "INDEL") %>%
  mutate(binary_response = factor(binary_response)) %>%
  make_boxplot_response(main = "Riaz : INDEL : MHC < 50")

p7 <- count50_expressed   %>%
  filter(cohort == "riaz_pre" & class == "INDEL") %>%
  mutate(binary_response = factor(binary_response)) %>%
  make_boxplot_response(main = "Riaz : INDEL : expressed + MHC < 50 ")

ggarrange(p1, p2, p3, p4, p5, p6, p7 , common.legend = TRUE,
          labels = c("C", "D", "E", "F", "G", "H", "I"),
          ncol = 3, nrow = 3)  

if(export_plots){
  dev.off()
}
```

### 1.3 Number of neoantigen candidates comparison before and after treament in Riaz cohort

Here, we compare the number of neoantigen candidates before and after ICB treatment in the NAIVE Riaz cohort. There are some clear trends that neoantigen candidate load decreases in responders while rather increases in non-responder but these effects are not significant for any mutation class. The following plots show all predicted neoantigen candidates. However, a similar non-significant trend could be observed when considering MHC binding. 

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# # merge pre and on samples
# # NOTE: many samples were excluded for the following reasons: no WES/RNA-seq, progressed already on previous ICB treatment, NE as response
# 
riaz_merged <- df_resp_numbers$riaz_pre %>%
  left_join(df_resp_numbers$riaz_on, by = "id", suffix = c("_pre", "_on"))

riaz_merged <- df_resp_numbers_expressed$riaz_pre %>%
  left_join(df_resp_numbers_expressed$riaz_on, by = "id", suffix = c("_pre", "_on"))

# 
riaz_merged_resp <-   riaz_merged %>% filter(binary_response_pre == "yes")
riaz_merged_noresp <-   riaz_merged %>% filter(binary_response_pre == "no")


p1 <- ggpaired(riaz_merged_resp, cond2 = "snv_on" , cond1 = "snv_pre", title = "responder")
p2 <- ggpaired(riaz_merged_noresp, cond2 = "snv_on" , cond1 = "snv_pre", title = "non-responder")
grid.arrange(p1, p2, ncol=2)

p1 <- ggpaired(riaz_merged_resp, cond2 = "INDEL_on" , cond1 = "INDEL_pre", title = "responder")
p2 <-ggpaired(riaz_merged_noresp, cond2 = "INDEL_on" , cond1 = "INDEL_pre", title = "non-responder")
grid.arrange(p1, p2, ncol=2)

p1 <- ggpaired(riaz_merged_resp, cond2 = "fusion gene_on" , cond1 = "fusion gene_pre", title = "responder")
p2 <- ggpaired(riaz_merged_noresp, cond2 = "fusion gene_on" , cond1 = "fusion gene_pre", title = "non-responder")
grid.arrange(p1, p2, ncol=2)

wilcox.test(riaz_merged_resp$snv_pre, riaz_merged_resp$snv_on, paired = T)
wilcox.test(riaz_merged_noresp$snv_pre, riaz_merged_noresp$snv_on, paired = T)

wilcox.test(riaz_merged_resp$INDEL_pre, riaz_merged_resp$INDEL_on, paired = T)
wilcox.test(riaz_merged_noresp$INDEL_pre, riaz_merged_noresp$INDEL_on, paired = T)

wilcox.test(riaz_merged_resp$`fusion gene_pre`, riaz_merged_resp$`fusion gene_on`, paired = T)
wilcox.test(riaz_merged_noresp$`fusion gene_pre`, riaz_merged_noresp$`fusion gene_on`, paired = T)


```


## Neoantigen quality as definded by NeoFox features

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

features.nummeric <- features_full

plot_density_mutation <- function(feature, snv, indel, fusion) {
  
  class_rename <- c("snv" = "SNV", "indel"= "INDEL", "fusion"= "Fusion gene")
  
  tsnv <-
    tibble(val = snv[[feature]],
           class = "snv")
  tindel <-
    tibble(val = indel[[feature]],
           class = "indel")
  tfusion <-
    tibble(val = fusion[[feature]],
           class = "fusion")
  
  t <- bind_rows(tsnv, tindel, tfusion)
  t <- t %>%
    mutate(class = class_rename[class]) %>%
    mutate(class = factor(class, levels = c("SNV", "INDEL", "Fusion gene")))
  
  ggplot(t, aes(val, fill = class)) + geom_density(alpha = 0.9) + scale_x_log10() +  xlab(feature) + scale_fill_manual(values=col_mutation)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank())
  
}

pls <- lapply(features.nummeric, function(x){
  plot_density_mutation(feature = x, snv = dat_transformed_comb$all_cohorts, indel = dat_indel_transformed_comb$all_cohorts,
                      fusion = dat_fusions_transformed_comb$all_cohorts)
})

if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_density_features.pdf")
  #pdf(nam)
  ggsave(nam, marrangeGrob(grobs = pls, nrow=2, ncol=2))
  #dev.off()
}else{
  pls
}


```


## 2 ROC-AUC for quantity

### 2.1 RCC + MEL

#### 2.1.1 SNV

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# SNVS
DF2_resp_snv <- DF2_resp %>%
  filter(class == "SNV") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))


DF2_resp_snv_folds <- nested_CV(DF2_resp_snv)
predictions_full <- lapply(DF2_resp_snv_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_snv_folds, function(y) y$binary_response)
auc_snvs <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


#### 2.1.2 Combined

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_combined <- DF2_resp %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_combined_folds <- nested_CV(DF2_resp_combined)
predictions_full <- lapply(DF2_resp_combined_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_combined_folds, function(y) y$binary_response)
auc_combined <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


#### 2.1.3 Fusions

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_fusions <- DF2_resp %>%
  filter(class == "Fusion gene") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_fusions_folds <- nested_CV(DF2_resp_fusions)
predictions_full <- lapply(DF2_resp_fusions_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_fusions_folds, function(y) y$binary_response)
auc_fusions <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


#### 2.1.4 INDELS

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_indel <- DF2_resp %>%
  filter(class == "INDEL") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_indel_folds <- nested_CV(DF2_resp_indel)
predictions_full <- lapply(DF2_resp_indel_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_indel_folds, function(y) y$binary_response)
auc_indel <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


### 2.2 MEL

#### 2.2.1 SNV

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# SNVS
DF2_resp_snv <- DF2_resp %>%
  filter(cohort.x %in% c("Van Allen","Hugo", "Riaz" ))%>%
  filter(class == "SNV") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))


DF2_resp_snv_folds <- nested_CV(DF2_resp_snv)
predictions_full <- lapply(DF2_resp_snv_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_snv_folds, function(y) y$binary_response)
auc_snvs_mel <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```

#### 2.2.2 Combined

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_combined <- DF2_resp %>%
  filter(cohort.x %in% c("Van Allen","Hugo", "Riaz" ))%>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_combined_folds <- nested_CV(DF2_resp_combined)
predictions_full <- lapply(DF2_resp_combined_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_combined_folds, function(y) y$binary_response)
auc_combined_mel <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)

```


#### 2.2.3 Fusions

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_fusions <- DF2_resp %>%
  filter(cohort.x %in% c("Van Allen","Hugo", "Riaz" ))%>%
  filter(class == "Fusion gene") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_fusions_folds <- nested_CV(DF2_resp_fusions)
predictions_full <- lapply(DF2_resp_fusions_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_fusions_folds, function(y) y$binary_response)
auc_fusions_mel <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


#### 2.2.4 INDELS

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_indel <- DF2_resp %>%
  filter(cohort.x %in% c("Van Allen","Hugo", "Riaz" ))%>%
  filter(class == "INDEL") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_indel_folds <- nested_CV(DF2_resp_indel)
predictions_full <- lapply(DF2_resp_indel_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_indel_folds, function(y) y$binary_response)
auc_indel_mel <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


### 2.3 RCC
#### 2.3.1 SNV

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# SNVS
DF2_resp_snv <- DF2_resp %>%
  filter(cohort.x %in% c("Miao","McDermott" ))%>%
  filter(class == "SNV") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))


DF2_resp_snv_folds <- nested_CV(DF2_resp_snv)
predictions_full <- lapply(DF2_resp_snv_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_snv_folds, function(y) y$binary_response)
auc_snvs_rcc <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```

#### 2.3.2 Combined

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# SNVS
DF2_resp_combined <- DF2_resp %>%
  filter(cohort.x %in% c("Miao","McDermott" ))%>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_combined_folds <- nested_CV(DF2_resp_combined)
predictions_full <- lapply(DF2_resp_combined_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_combined_folds, function(y) y$binary_response)
auc_combined_rcc <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```


#### 2.3.3 Fusions

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_fusions <- DF2_resp %>%
  filter(cohort.x %in% c("Miao","McDermott" ))%>%
  filter(class == "Fusion gene") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_fusions_folds <- nested_CV(DF2_resp_fusions)
predictions_full <- lapply(DF2_resp_fusions_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_fusions_folds, function(y) y$binary_response)
auc_fusions_rcc <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```



#### 2.3.4 INDELS

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

DF2_resp_indel <- DF2_resp %>%
  filter(cohort.x %in% c("Miao","McDermott" ))%>%
  filter(class == "INDEL") %>%
  filter(!is.na(binary_response))%>%
  filter(!is.na(count))

DF2_resp_indel_folds <- nested_CV(DF2_resp_indel)
predictions_full <- lapply(DF2_resp_indel_folds, function(y) y$count)
binary_response <- lapply(DF2_resp_indel_folds, function(y) y$binary_response)
auc_indel_rcc <- roc_with_cv(predictions_full = predictions_full, labels = binary_response)


```

#### 2.4 Summary 

```{r , echo=FALSE, message=FALSE, warning = FALSE, results=FALSE}

# summary of quantitative model 
auc_rcc_mel <- bind_rows(list(SNV = auc_snvs, "Fusion gene" = auc_fusions, "INDEL" = auc_indel, "combined" = auc_combined), .id = "class")
auc_mel <- bind_rows(list(SNV = auc_snvs_mel, "Fusion gene" = auc_fusions_mel, "INDEL" = auc_indel_mel, "combined" = auc_combined_mel), .id = "class")
auc_rcc <- bind_rows(list(SNV = auc_snvs_rcc, "Fusion gene" = auc_fusions_rcc, "INDEL" = auc_indel_rcc, "combined" = auc_combined_rcc), .id = "class")
auc_overall <- bind_rows(list("MEL+RCC" = auc_rcc_mel, "MEL" = auc_mel, "RCC" = auc_rcc), .id = "entity")



if(export_plots){
  nam <- paste0(path_to_plots, "/", date_today, "_ICB_quantitative_model_bar_auc_nested_cv_without_SD.pdf")
  pdf(nam, height = 6, width = 10)
}
auc_overall %>%
  mutate(class = factor(class, levels = c("SNV", "INDEL", "Fusion gene", "combined"))) %>%
  mutate(entity = factor(entity, levels = c("MEL+RCC", "MEL", "RCC"))) %>%
  ggplot(aes(class, mean_auc, fill = entity))+
  geom_bar(stat = "identity", position = "dodge")+
  geom_hline(yintercept = 0.5) +
  geom_errorbar(aes(ymin=mean_auc-sd_auc , ymax=mean_auc+sd_auc ), width=.2,
                 position=position_dodge(.9), color = "darkgrey") +
  scale_fill_manual(values = rev(col_entity))

if(export_plots){
  dev.off()
}


if(export_table){
  nam <- paste0("data_for_publication/",date_today, "_quantitative_model_mean_auc_nested_cv_wo_SD.txt")
  write_delim(auc_overall, nam)
}


```

